server/src/server.rs

```diff
@@ -298,25 +334,125 @@ impl Default for Settings {
 			enable_ws: true,
 			message_buffer_capacity: 1024,
 			ping_config: PingConfig::WithoutInactivityCheck(Duration::from_secs(60)),
+			id_provider: Arc::new(RandomIntegerIdProvider),
+		}
+	}
+}
+
+impl ServerConfig {
+	/// Create a new builder for the [`ServerConfig`].
+	pub fn builder() -> ServerConfigBuilder {
+		ServerConfigBuilder::default()
+	}
+}
+
+impl Default for ServerConfigBuilder {
+	fn default() -> Self {
+		let this = ServerConfig::default();
+
+		ServerConfigBuilder {
+			max_request_body_size: this.max_request_body_size,
+			max_response_body_size: this.max_response_body_size,
+			max_connections: this.max_connections,
+			max_subscriptions_per_connection: this.max_subscriptions_per_connection,
+			batch_requests_config: this.batch_requests_config,
+			enable_http: this.enable_http,
+			enable_ws: this.enable_ws,
+			message_buffer_capacity: this.message_buffer_capacity,
+			ping_config: this.ping_config,
+			id_provider: this.id_provider,
+		}
+	}
+}
+
+impl ServerConfigBuilder {
+	/// Create a new [`ServerConfigBuilder`].
+	pub fn new() -> Self {
+		Self::default()
+	}
+
+	/// See [`Builder::max_request_body_size`](method@Builder::max_request_body_size) for documentation.
+	pub fn max_request_body_size(mut self, size: u32) -> Self {
```

Comment:
this is duplicated :(

we could add a trait for but it doesn't seems worth it for now